### PR TITLE
Lau task deletion

### PR DESCRIPTION
**Pull request by priyankaVerma21**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/LAU-545

### Change description ###
This change is for task deletion

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by MartinYSpasov** [comment]
Closing PR, we need to merge WA changes first before we merge

**Comment by MartinYSpasov** [comment]
Closing until Priyanka fix all remaining issues.

File: build.gradle [hunk 1]
```diff
@@ -186,7 +186,7 @@ dependencyManagement {
             entry 'spring-cloud-starter-openfeign'
             entry 'spring-cloud-openfeign-core'
         }
-        dependencySet(group: 'org.yaml', version: '2.2') {
+        dependencySet(group: 'org.yaml', version: '1.33') {
```
Comment (MartinYSpasov) [inline build.gradle:189]: This probably happened during the rebase yesterday, please use v2.2

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/utils/TaskDeleteTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import static org.awaitility.Awaitility.with;
+
+@Component
+public class TaskDeleteTestUtils {
+
+    @Inject
+    private TasksDeletionRecordHolder tasksDeletionRecordHolder;
+
+    @Inject
+    private SecurityUtils securityUtils;
```
Comment (MartinYSpasov) [inline src/functionalTest/java/uk/gov/hmcts/reform/ccd/utils/TaskDeleteTestUtils.java:23]: I don't see any usage of securityUtils, parameterResolver, s2SHelper and fileUtils

File: src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@Getter
+public class TasksDeletionRecordHolder {
+    private List<Map<String, Integer>> tasksDeletionRecordHolderList = new ArrayList<>();
+
+    public void setCaseTasksDeletionResults(final String caseRef,
+                                            final int caseTasksDeletionResults) {
+        tasksDeletionRecordHolderList.add(Map.of(caseRef, caseTasksDeletionResults));
+    }
+
+    //TODO: this looks ridiculous :)
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java:21]: Please remove my ridiculous comment :)

File: src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    public int getTasksDeletionResults(final String caseRef) {
+        if (!tasksDeletionRecordHolderList.isEmpty()) {
+            final Optional<Map<String, Integer>> deletionResultsMap =
+                tasksDeletionRecordHolderList.stream()
+                    .filter(roleHolderEntry -> roleHolderEntry.containsKey(caseRef))
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java:26]: this is not roleHolderEntry

File: src/integrationTest/java/uk/gov/hmcts/reform/ccd/config/WireMockStubs.java [hunk 5]
```diff
@@ -31,6 +34,8 @@ public class WireMockStubs {
     private static final String ROLES_QUERY_PATH = "/am/role-assignments/query";
     private static final String LAU_SAVE_PATH = "/audit/caseAction";
 
+    public static final String TASKS_DELETE_PATH = "/task/delete";
```
Comment (MartinYSpasov) [inline src/integrationTest/java/uk/gov/hmcts/reform/ccd/config/WireMockStubs.java:37]: The string should be private

File: config/owasp/suppressions.xml [hunk 6]
```diff
@@ -11,7 +11,15 @@
   </suppress>
   <suppress>
     <packageUrl regex="true">^pkg:maven/com.massisframework/j-text-utils@.*$</packageUrl>
-    <cve>CVE-2023-2972</cve>
+    <cve>CVE-2021-4277</cve>
+  </suppress>
+  <suppress>
+    <packageUrl regex="true">^pkg:maven/org.yaml/snakeyaml@.*$</packageUrl>
```
[the remark this answers is not in the document]
Comment (priyankaVerma21) [reply config/owasp/suppressions.xml:17]: Hi @MartinYSpasov , This is now inline with master.

File: src/main/java/uk/gov/hmcts/reform/ccd/data/tm/DeleteCaseTasksAction.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+@ToString
+public class DeleteCaseTasksAction {
```
[the remark this answers is not in the document]
Comment (priyankaVerma21) [reply src/main/java/uk/gov/hmcts/reform/ccd/data/tm/DeleteCaseTasksAction.java]: This is now implementing Serailizable

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/utils/TaskDeleteTestUtils.java (hunk 2 above)
Comment (priyankaVerma21) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/utils/TaskDeleteTestUtils.java]: These are now removed

File: src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java (hunk 3 above)
Comment (priyankaVerma21) [reply src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java]: This has now been removed

File: src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java (hunk 4 above)
Comment (priyankaVerma21) [reply src/main/java/uk/gov/hmcts/reform/ccd/util/log/TasksDeletionRecordHolder.java]: This has been changed

**Review by MartinYSpasov — APPROVED** [review]

**Pull request merged**